Cargo.toml

```diff
@@ -391,13 +393,27 @@ zeroize = { version = "1.6.0", features = ["zeroize_derive", "std"] }
 zip = { version = "0.6.6", default-features = false, features = ["deflate","bzip2"] }
 zone = { version = "0.3", default-features = false, features = ["async"] }
 
+# NOTE: The test profile inherits from the dev profile, so settings under
+# profile.dev get inherited. AVOID setting anything under profile.test: that
+# will cause dev and test builds to diverge, which will cause more Cargo build
+# cache misses.
```

Comment:
Good catch, and AFAICT, that's a large part of the rest of this PR, correct? Ensuring that "build for dev" and "build for test" are aligned, so they only build once?

Reply:
Yes, the biggest chunk by loc is removing divergence between dev and test builds.